python/lib/eeg.py

```diff
@@ -238,262 +227,237 @@ def grep_bids_files(self, bids_type):
-                    self.fetch_and_insert_event_file(eeg_file_id, derivatives)
 
-    def fetch_and_insert_eeg_file(self, derivatives=None):
+    def fetch_and_insert_eeg_files(self, derivatives=False):
         """
         Gather EEG file information to insert into physiological_file and
         physiological_parameter_file. Once all the information has been
         gathered, it will call self.insert_physiological_file that will
         perform the insertion into physiological_file and
         physiological_parameter_file.
 
-        :param derivatives: dictionary with derivative folder information if
-                            the EEG file to insert is a derivative file.
-                            Set by default to None when inserting raw file.
-         :type derivatives: list
+        :param derivatives: True if the EEG file to insert is a derivative file.
+                            Set by default to False when inserting raw file.
+         :type derivatives: boolean
         :return: dictionary with registered file ID and path to its file
          :rtype: dict
         """
 
+        inserted_eegs = []
         # load the Physiological object that will be used to insert the
         # physiological data into the database
         physiological = Physiological(self.db, self.verbose)
 
-        # check if inserting derivatives to use the derivative_pattern to
-        # grep for the eeg file
-        derivative_pattern = None
-        derivative_path    = None
-        files_list         = self.eeg_files  # by default, raw data is eeg_files
-        if derivatives:
-            # TODO grep the source file as well as the input file ID???
-            derivative_pattern = derivatives['derivative_name'] + "/sub-"
-            files_list         = self.derivative_eeg_files
-            derivative_path    = self.get_derivatives_path(derivatives)
-
-        # grep the raw files from eeg_files list
-        eeg_file = BidsReader.grep_file(
-            files_list         = files_list,
-            match_pattern      = ".(set$|edf$|vhdr$|vmrk$|eeg$|bdf$)",
-            derivative_pattern = derivative_pattern
-        )
-        json_file = BidsReader.grep_file(
-            files_list         = files_list,
-            match_pattern      = '.json$',
-            derivative_pattern = derivative_pattern
-        )
-        fdt_file = BidsReader.grep_file(
-            files_list         = files_list,
-            match_pattern      = '.fdt$',
-            derivative_pattern = derivative_pattern
+        # TODO if derivatives, grep the source file as well as the input file ID???
+
+        # grep the raw files
+        eeg_files = self.bids_layout.get(
+            subject   = self.bids_sub_id,
+            session   = self.bids_ses_id,
+            scope     = 'derivatives' if derivatives else 'raw',
+            #datatype = self.bids_modality,
+            #suffix   = self.bids_modality,
```

Comment:
why are those two commented?

Reply:
Those line mimic the previous logic, but are commented because I think it may be good to not be too restrictive. You can let me know your though and I will adjust accordingly.

Reply:
I would tend to keep them to ensure you are getting only the EEG or iEEG files. For example, maybe for MEG or other electrophysiological recordings, the files will have the same extensions as EEG but their insertion process might be different than EEG? 
At least, we are sure we will not get the MRIs since the extension of images are `.nii.gz` but relying only on file extensions to grep the proper files seem risky. What do you think?

Reply:
Given your reasoning, I agree with you. Fixed!